src/main/webapp/app/shared/metis/posting.directive.ts

```diff
@@ -1,18 +1,71 @@
 import { Posting } from 'app/entities/metis/posting.model';
-import { Directive, Input, OnInit } from '@angular/core';
+import { ChangeDetectorRef, Directive, Input, OnDestroy, OnInit, inject } from '@angular/core';
+import { MetisService } from 'app/shared/metis/metis.service';
 
 @Directive()
-export abstract class PostingDirective<T extends Posting> implements OnInit {
+export abstract class PostingDirective<T extends Posting> implements OnInit, OnDestroy {
     @Input() posting: T;
     @Input() isCommunicationPage: boolean;
     @Input() showChannelReference?: boolean;
 
     @Input() hasChannelModerationRights = false;
     @Input() isThreadSidebar: boolean;
 
+    isAnswerPost = false;
+    isDeleted = false;
+    readonly timeToDeleteInSeconds = 6;
+    deleteTimerInSeconds = 6;
+    deleteTimer: NodeJS.Timeout | undefined;
+    deleteInterval: NodeJS.Timeout | undefined;
+
     content?: string;
 
+    protected metisService = inject(MetisService);
+    protected changeDetector = inject(ChangeDetectorRef);
+
     ngOnInit(): void {
         this.content = this.posting.content;
     }
+
+    ngOnDestroy(): void {
+        if (this.deleteTimer !== undefined) {
+            clearTimeout(this.deleteTimer);
+        }
+
+        if (this.deleteInterval !== undefined) {
+            clearInterval(this.deleteInterval);
+        }
+    }
+
+    onDeleteEvent(isDelete: boolean) {
+        this.isDeleted = isDelete;
+
+        if (this.deleteTimer !== undefined) {
+            clearTimeout(this.deleteTimer);
+        }
+
+        if (this.deleteInterval !== undefined) {
+            clearInterval(this.deleteInterval);
+        }
+
+        if (isDelete) {
+            this.deleteTimerInSeconds = this.timeToDeleteInSeconds;
+
+            this.deleteTimer = setTimeout(
+                () => {
+                    if (this.isAnswerPost) {
+                        this.metisService.deleteAnswerPost(this.posting);
+                    } else {
+                        this.metisService.deletePost(this.posting);
+                    }
+                },
+                this.deleteTimerInSeconds * 1000 + 1000,
```

Comment:
Why the + 1000?
Maybe add a comment

Reply:
Added a comment in this [commit](https://github.com/ls1intum/Artemis/pull/9624/commits/82a35634c5635a96949b32342c0bee087f8da25a).